File: LED_control/sw1.c
//LED 켜고 끄는 것을 스위치로 제어

#include <stdio.h>
#include <wiringPi.h>
#define LED1 4
#define SW1 1

int main(void){
    unsigned int i = 0;
    if(wiringPiSetup() == -1) return 1;
    pinMode(LED1,OUTPUT);
    pinMode(SW1, INPUT);
    pinMode(SW1, PUD_DOWN); 
    //pull down 하면 스위치가 눌렷을 때 1, 안눌리면 0
    for(;;){
        if(digitalRead(SW1) == 1){
            if(i==0) i==1;
            else i==0;
        }
        digitalWrite(LED1, i);
    }
    return 0;
}

File: LED_control/sw1.py
import RPi.GPIO as GPIO
GPIO.setmode(GPIO.BOARD)

GPIO.setup(12,GPIO.IN,pull_up_down=GPIO.PUD_DOWN)
GPIO.setup(16,GPIO.OUT)

i=0
while True:
    if(GPIO.input(12)==1):
        if i==0: i=1
        else: i=0
    GPIO.output(16,i)
    
GPIO.cleanup()
